### PR TITLE
lang: Add file function for reading files

### PR DESCRIPTION
**Pull request by aequitas**
Todos:

- [x] streaming updates
- [ ] better error handling?
- [x] tests
- [ ] https://github.com/purpleidea/mgmt/pull/365/files#diff-a7d4a5bd50bf5e859cbd3bb62e51b65cR92 pending on fix for: https://github.com/purpleidea/mgmt/pull/367

### CODE REVIEW

**Comment by aequitas** [comment]
- [ ] test

File: lang/funcs/core/file_func.go [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+//
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+package core // TODO: should this be in its own individual package?
+
+import (
+	"io/ioutil"
+
+	errwrap "github.com/pkg/errors"
```
Comment (purpleidea) [inline lang/funcs/core/file_func.go]: TestUnification1

Comment (aequitas) [reply lang/funcs/core/file_func.go]: this is what goimports makes of it, can we put this rule into a config file for goimports? I really don't want to be managing imports (and their sorting) by hand. A computer can do this just fine :)

Comment (purpleidea) [reply lang/funcs/core/file_func.go]: ¯\_(ツ)_/¯

I thought it did do this. If not, it's a bug, missing feature, or a property not explicitly enforced.
It is the common approach we use here and in is seen in many other projects too.

**Comment by aequitas** [comment]
Rebase on latest master, made all test function `t.Helper` functions so they provide a better trace on a `t.Fatal`.

**Comment by aequitas** [comment]
https://github.com/purpleidea/mgmt/pull/368#issuecomment-372639491

**Pull request closed without merging**